packages/analytics-js-plugins/__tests__/deviceModeDestinations/index.test.ts

```diff
@@ -381,6 +381,287 @@ describe('DeviceModeDestinations Plugin', () => {
+        const mixedDestinations: Destination[] = [
+          {
+            ...mockDestinations[0]!,
+            enabled: true, // Should be included
+          },
+          {
+            ...mockDestinations[1]!,
+            enabled: false, // Should be filtered out
+          },
+          {
+            id: 'dest4',
+            displayName: 'GA4',
+            userFriendlyId: 'GA4___dest4',
+            enabled: false, // Should be filtered out
+            shouldApplyDeviceModeTransformation: true,
+            propagateEventsUntransformedOnError: false,
+            config: {
+              apiKey: 'key4',
+              blacklistedEvents: [],
+              whitelistedEvents: [],
+              eventFilteringOption: 'disable' as const,
+            },
+          },
+        ];
+
+        mockState.nativeDestinations.configuredDestinations.value = mixedDestinations;
+        mockState.loadOptions.value = {};
+        mockState.nativeDestinations.loadOnlyIntegrations.value = { All: true };
+
+        // Mock filterDisabledDestination to actually filter (since we're testing the real behavior)
+        mockApplySourceConfigurationOverrides.mockImplementation(destinations =>
+          destinations.filter(dest => dest.enabled),
+        );
+
+        plugin.nativeDestinations.setActiveDestinations(
+          mockState,
+          mockPluginsManager,
+          mockErrorHandler,
+          mockLogger,
+        );
+
+        // Verify only enabled destinations are passed to filterDestinations
+        expect(mockFilterDestinations).toHaveBeenCalledWith(
+          { All: true },
+          expect.arrayContaining([expect.objectContaining({ id: 'dest1', enabled: true })]),
+        );
+
+        // Verify disabled destinations are not included
+        const passedDestinations = mockFilterDestinations.mock.calls[0]?.[1];
+        expect(passedDestinations).not.toContainEqual(expect.objectContaining({ enabled: false }));
+        expect(passedDestinations).toHaveLength(1);
+        expect(passedDestinations?.[0]).toEqual(
+          expect.objectContaining({ id: 'dest1', enabled: true }),
+        );
+      });
+
+      it('should filter disabled destinations even when load options explicitly include them', () => {
+        const mixedDestinations: Destination[] = [
+          {
+            ...mockDestinations[0]!,
+            displayName: 'GA4',
+            enabled: true,
+          },
+          {
+            ...mockDestinations[1]!,
+            displayName: 'Google Analytics',
+            enabled: false, // Disabled destination
+          },
+        ];
+
+        mockState.nativeDestinations.configuredDestinations.value = mixedDestinations;
+        mockState.loadOptions.value.sourceConfigurationOverride = undefined;
+
+        // Load options that would normally include both destinations
+        mockState.nativeDestinations.loadOnlyIntegrations.value = {
+          All: false,
+          GA4: true,
```

Comment:
The correct name is "Google Analytics 4 (GA4)". Let's fix the mock for the names map.

Reply:
Addressed